packages/trpc/server/profile-router/router.ts

```diff
@@ -110,4 +111,15 @@ export const profileRouter = router({
       });
     }
   }),
+  isPasswordNullUser: procedure.query(async ({ ctx }) => {
+    try {
+      const isUserPasswordNull = await isPasswordNull({ id: ctx.user?.id as number });
+      return isUserPasswordNull;
+    } catch (err) {
+      throw new TRPCError({
+        code: 'BAD_REQUEST',
+        message: 'An error occurred while checking the user existence. Please try again.',
+      });
+    }
+  }),
```

Comment:
We can probably just retrieve this during the server render and eliminate the need for the tRPC endpoint.

That said I do appreciate that you're just using the ctx here for the userId and not introducing it as a param which could be abused, good work!

Reply:
Do you mean creating the Next.js server component to retrieve data directly?